packages/@n8n/nodes-langchain/nodes/tools/ToolThink/ToolThink.node.ts

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable n8n-nodes-base/node-dirname-against-convention */
+import { Tool } from '@langchain/core/tools';
+import {
+	NodeConnectionTypes,
+	type INodeType,
+	type INodeTypeDescription,
+	type ISupplyDataFunctions,
+	type SupplyData,
+} from 'n8n-workflow';
+
+import { logWrapper } from '@utils/logWrapper';
+import { getConnectionHintNoticeField } from '@utils/sharedFields';
+
+// A thinking tool, see https://www.anthropic.com/engineering/claude-think-tool
+interface ThinkToolParams {
+	subject: string;
+}
+
+class ThinkTool extends Tool {
+	static lc_name(): string {
+		return 'ThinkTool';
+	}
+
+	name = 'think';
+
+	description =
+		'Use the tool to think about something. It will not obtain new information or change the database, but just append the thought to the log. Use it when complex reasoning or some cache memory is needed.';
+
```

Comment:
Perhaps it would make sense to allow users to change this default description via parameter? 🤔